### DISCUSSION TITLE
Disable registration on selfhosted server

### DISCUSSION THREAD

**Discussion by pinpox** [ente-io/ente#2476] [Q&A] [answered]
Hello,
is it possible to disable registration on my selfhosted server?
I would like to only have the users that I (the admin) have created

**Comment by mnvr** [comment, marked as answer]
It is not possible currently. However, if you've not configured email (or disable it after you've created your users), then there will not be a way for people to register since they cannot get the OTP.

Alternatively, once you have registered all the users you need, you can then disable registration by re-deploying museum after commenting out the following line - https://github.com/ente-io/ente/blob/f2057cbaa895ecc8df7419d97f570811e23eec67/server/pkg/controller/user/userauth.go#L324

This will stop new accounts from being created.

**Reply by pinpox** [reply]
Thanks for the information! While that mostly works, a proper setting would be nice.

Are ther any plans of adding a setting for it or ideas on how it would be implemented? 
Consider this a feature request :)

**Reply by mnvr** [reply]
> Consider this a feature request :)

Got it :)

**Reply by neferin12** [reply]
> Alternatively, once you have registered all the users you need, you can then disable registration by re-deploying museum after commenting out the following line -
> 
> https://github.com/ente-io/ente/blob/f2057cbaa895ecc8df7419d97f570811e23eec67/server/pkg/controller/user/userauth.go#L324
> 
> This will stop new accounts from being created.

Would you consider accepting a pull request that makes the registration of a user configurable via the yaml file/env variable, to disable registration without needing to edit the code?

**Reply by mnvr** [reply]
Okay. Let us add "internal.disable-registration". Thank you.

**Reply by pinpox** [reply]
It might be a good idea to add a flag/envvar for the webui aswell to disable the link for registration when set. Something similar like the existing overrides for the endpoint

**Reply by mnvr** [reply]
Not sure about that. Web is not the only client.

Let's keep the scope small and only deal with the museum bit for now.
